ssl/ssl_sess.c

```diff
@@ -78,7 +80,7 @@ SSL_SESSION *SSL_SESSION_new(void)
     ss->verify_result = 1;      /* avoid 0 (= X509_V_OK) just in case */
     ss->references = 1;
     ss->timeout = 60 * 5 + 4;   /* 5 minute timeout by default */
-    ss->time = (unsigned long)time(NULL);
+    ss->time = (long)time(NULL);
```

Comment:
We use `time_t` already and that's what `time(2)` should return.
If possible, drop the cast here and elsewhere and type variables & fields `time_t`.

Reply:
Yeah, that might be a fairly invasive change... (not disagreeing with it, but this using legacy code)

Reply:
And the APIs currently use "long"... would that be an API breaker?

Reply:
Do we still want to use `time_t` here, even though it's originally defined as long; type change exposed to user?

Reply:
Sigh, legacy APIs.  We can't change the return or argument types of `SSL_SESSION_get_time()` and friends.

It would be possible to use `time_t` internally and use casts to cross the API boundary.  This would at least make the internal uses type safe.  I expect there are machines with 32 longs that have a longer `time_t`, or there will be at the rollover approaches.

Reply:
For 3.0 you can change API if you think it worthwhile.  The goal for 3.0 is mostly source compatible.

Reply:
@paulidale I can do the casts at API boundary... or I can change the APIs. Either way is fine by me.